packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -101,16 +121,60 @@ export default function MongoNav({
   admin = false,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
+
+  const accountHost = hosts?.account ?? `https://account.mongodb.com`;
+  const cloudHost = hosts?.cloud ?? `https://cloud.mongodb.com`;
+  const universityHost = hosts?.university ?? `https://university.mongodb.com`;
+  const supportHost = hosts?.support ?? `https://support.mongodb.com`;
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  function getProductionData() {
+    fetch(`${cloudHost}/user/shared`, {
+      credentials: 'include',
+      method: 'GET',
+    })
+      .then(response => response.json())
+      .then(data => {
+        setData(data);
+        onSuccess?.(data);
+      })
+      .catch(error => {
+        onError?.(error);
+        console.warn(error);
```

Comment:
`console.error` is more appropriate here.

Reply:
👍